### PR TITLE
Feature: Support Event Tracing when calling Aws S3 API

### PR DESCRIPTION
**Pull request by mateng0915**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Description
===========
In Computing && Storage Architecture, using S3 as remote / Shared storage, the Method to access S3 is using AWS S3 API
However there is a gap between clickhouse DBA and S3 SRE,
S3 SRE want to know the QPS is that S3 API be called, but in ClickHouse side
our operation is SQL and background task
And some SQL maybe need calling multi S3 APIs

Solution
========
So implemented this improved feature
Added S3 API event in the system.events, then using clickhouse-exporter
can grab this metric data and build the Monitor Dashboard

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Review by serxa — CHANGES_REQUESTED** [review]
Generally ok, but couple issues have to be fixed before merging (see review), also there is failed integration test `test_profile_events_s3/test.py::test_profile_events` with every profile event about s3. Please, add your events in that test.

**Comment by mateng0915** [comment]
> PutS3ObjectRequest



> Generally ok, but couple issues have to be fixed before merging (see review), also there is failed integration test `test_profile_events_s3/test.py::test_profile_events` with every profile event about s3. Please, add your events in that test.

Thx @serxa 's comments, already removed the unused event and change some increment event  into right logic function

**Comment by mateng0915** [comment]
> > PutS3ObjectRequest
> 
> > Generally ok, but couple issues have to be fixed before merging (see review), also there is failed integration test `test_profile_events_s3/test.py::test_profile_events` with every profile event about s3. Please, add your events in that test.
> 
> Thx @serxa 's comments, already removed the unused event , change some increment event into right logic function and added new event into integration test

File: src/Disks/S3/DiskS3.cpp [hunk 1]
```diff
@@ -0,0 +1,1147 @@
+#include "DiskS3.h"
```
Comment (serxa) [inline src/Disks/S3/DiskS3.cpp:1]: This file huge changes looks wrong. How this happened?

Comment (mateng0915) [reply src/Disks/S3/DiskS3.cpp]: many thx your review, it is my local file , already delete it.

**Comment by mateng0915** [comment]
@serxa  hello serxa, could  you help check the review comments when you have time , i already fixed them all. Thx bro

File: tests/integration/test_profile_events_s3/test.py [hunk 2]
```diff
@@ -56,6 +56,17 @@ def cluster():
...
+    "CopyS3Object": 0,
+    "ListS3Objects": 0,
+    "HeadS3Object": 0,
+    "CreateS3MultipartUpload": 0,
+    "UploadS3PartCopy": 0,
+    "UploadS3Part": 0,
+    "AbortS3MultipartUpload": 0,
+    "CompleteS3MultipartUpload": 0,
+    "PutS3ObjectRequest": 0,
+    "GetS3ObjectRequest" 0,
```
Comment (serxa) [inline tests/integration/test_profile_events_s3/test.py:69]:
```suggestion
    "GetS3ObjectRequest": 0,
```

Comment (serxa) [reply tests/integration/test_profile_events_s3/test.py:69]: There is style check failing, looks like this is the only place it does not like. You can also see more CI check, after style will be fixed.

**Comment by serxa** [comment]
LGTM. The only thing left is to make CI checks green

**Review by serxa — APPROVED** [review]

**Pull request merged**